eng/scripts/StartDumpCollectionForHangingBuilds.ps1

```diff
@@ -0,0 +1,112 @@
+param(
+  [Parameter(Mandatory = $true)]
+  [ValidateNotNullOrEmpty()]
+  [string]
+  $ProcDumpPath,
+  [Parameter(Mandatory = $true)]
+  [ValidateNotNullOrEmpty()]
+  [string]
+  $ProcDumpDumpFolderPath,
+  [Parameter(Mandatory = $true)]
+  [datetime]
+  $WakeTime,
+  [Parameter(Mandatory = $true)]
+  [ValidateNotNullOrEmpty()]
+  [string []]
+  $CandidateProcessNames
+)
+
+Write-Output "Setting up a scheduled job to capture process dumps.";
+
+if ((-not (Test-Path $ProcDumpPath))) {
+  Write-Warning "Can't find ProcDump at '$ProcDumpPath'.";
+}
+else {
+  Write-Output "Using ProcDump from '$ProcDumpPath'.";
+}
+
+try {
+  Get-Job -Name CaptureDumps* | Stop-Job -PassThru | Remove-Job;
+  Get-ScheduledJob CaptureDumps* | Unregister-ScheduledJob;
+}
+catch {
+  Write-Output "There was an error cleaning up previous jobs.";
+  Write-Output $_.Exception.Message;
+}
+
+$repoRoot = Resolve-Path "$PSScriptRoot\..\..";
+$ProcDumpDumpFolderPath = Join-Path $repoRoot $ProcDumpDumpFolderPath;
+
+Write-Output "Dumps will be placed at '$ProcDumpDumpFolderPath'.";
+Write-Output "Watching processes $($CandidateProcessNames -join ', ')";
+
+# This script registers as a scheduled job. This scheduled job executes after two hours.
+# When the scheduled job executes, it runs procdump on all alive processes whose name matches $CandidateProcessNames.
+# The dumps are placed in $ProcDumpDumpFolderPath
+# If the build completes sucessfully in less than two hours, a final step unregisters the job.
+
+# Create a unique identifier for the job name
```

Comment:
I've seen conflicts happen if the job doesn't get cleaned up properly.
Using a unique identifier avoids the issue.
We don't want the build to fail or skip capturing dumps for potential hanging builds just because a previous job didn't got cleaned up properly.
Id rather be err on the side of reliability here.